contracts/base/ModuleManager.sol

```diff
@@ -379,32 +382,43 @@ abstract contract ModuleManager is Storage, EIP712, IModuleManagerEventsAndError
     }
 
     /// @notice Checks if an enable mode signature is valid.
-    /// @param digest signed digest.
+    /// @param structHash data hash.
     /// @param sig Signature.
-    function _checkEnableModeSignature(bytes32 digest, bytes calldata sig) internal view {
+    function _checkEnableModeSignature(bytes32 structHash, bytes calldata sig) internal view returns (bool) {
         address enableModeSigValidator = address(bytes20(sig[0:20]));
         if (!_isValidatorInstalled(enableModeSigValidator)) {
             revert ValidatorNotInstalled(enableModeSigValidator);
         }
-
-        if (IValidator(enableModeSigValidator).isValidSignatureWithSender(address(this), digest, sig[20:]) != ERC1271_MAGICVALUE) {
-            revert EnableModeSigError();
+        bytes32 eip712Digest = _hashTypedData(structHash);
+        // have to try different ways to validate the signature as we do not know for sure what 1271 flavours validator supports
+        if (IERC7739(enableModeSigValidator).supportsNestedTypedDataSign() == SUPPORTS_NESTED_TYPED_DATA_SIGN) {
+            // if the validator supports 7739, we use just the struct hash, as the full hash will be rebuilt inside 7739 flow
+            if (IValidator(enableModeSigValidator).isValidSignatureWithSender(address(this), eip712Digest, sig[20:]) == ERC1271_MAGICVALUE) {
+                return true;
+            }
+        } else {
+            // if the validator doesn't support 7739 for standard isValidSignatureWithSender, we provide the eip 712 digest
+            if (IValidator(enableModeSigValidator).isValidSignatureWithSender(address(this), eip712Digest, sig[20:]) == ERC1271_MAGICVALUE) {
+                return true;
```

Comment:
are these two code blocks meant to be identical?

Reply:
nope. great catch. I just noticed this today morning and fixed in #169